bindings/rust/wasmedge-sys/src/vm.rs

```diff
@@ -500,6 +541,25 @@ impl Vm {
         self.run_function_async(func_name, params)
     }
 
+    pub async fn run_wasm_from_module_async2(
+        &self,
+        module: Module,
+        func_name: impl AsRef<str> + Send,
+        params: impl IntoIterator<Item = WasmValue> + Send,
+    ) -> WasmEdgeResult<Vec<WasmValue>> {
+        // load
+        self.load_wasm_from_module(&module)?;
+
+        // validate
+        self.validate()?;
+
+        // instantiate
+        self.instantiate()?;
+
+        // invoke
+        self.run_function_async2(func_name, params).await
+    }
+
```

Comment:
Please replace the origin `run_wasm_from_module_async` with the new `run_wasm_from_module_async2`, but keep using the original API name `run_wasm_from_module_async`. Thanks!